### PR TITLE
New Roundstart Perks

### PR DESCRIPTION
**Pull request by Guidesu**
## About The Pull Request
Several Round start perks that are usable from the loadout (2 each so you can select 5 at maximum before using all of your pool of points... which could be used into better things like the spear or sword.....) It is literally a worse version of all the racial perks but to a big degree and some even reworked to be into its own niche. And Like herbivores and carnivores as well for species that can't get normally. You need to aim at your eyes with nothing covered to use them and apply it. Throw the thing after at the trash.

<details>
<summary>
	About The Pull Request
</summary>
<hr>

To be made in another PR:
//#define PERK_COLD_BLOODED /datum/perk/
//#define PERK_HOT_BLOODED /datum/perk/
//#define PERK_FAST_METABOLISM /datum/perk/
//#define PERK_SLOW_METABOLISM /datum/perk/
//#define PERK_NON_CONDUCTIVE /datum/perk/
//#define PERK_RENDING_CLAWS /datum/perk
//#define PERK_NITROGEN_EXHALE /datum/perk/
//#define PERK_NIGHT_SIGHT /datum/perk/
//#define PERK_PHOTO_RESISTANCE /datum/perk/
//#define PERK_AUTO_HISS /datum/perk/
//#define PERK_CONDUCTIVE /datum/perk/
//#define PERK_NO_DARK_SIGHT /datum/perk/

	
<hr>
</details>

## Changelog
:cl:

/:cl:

### CODE REVIEW

**Comment by Guidesu** [comment]
And FBPs can't gain this perks;......

**Comment by elgeonmb** [comment]
"Genetical" means "relating to genesis", which is not what you mean.
You want "Genetic", meaning "relating to genes".

File: code/modules/mob/living/carbon/human/human_recoil.dm [hunk 1]
```diff
@@ -17,5 +17,7 @@
 		base_recoil -= 0.4 //less recoil if your a good shot
 	if(usr.stats.getPerk(PERK_SMARTLINK))
 		base_recoil -= 0.2 //Worse than kriosans but probably enough to matter
+		if(usr.stats.getPerk(PERK_STRONGARMS))
+			base_recoil = 0.3 //Just a little
```
Comment (elgeonmb) [inline code/modules/mob/living/carbon/human/human_recoil.dm:21]: is this meant to be a setter and not subtraction like the other two perks in this family?

File: code/modules/mob/living/carbon/human/human_recoil.dm [hunk 2]
```diff
@@ -17,5 +17,7 @@
 		base_recoil -= 0.4 //less recoil if your a good shot
 	if(usr.stats.getPerk(PERK_SMARTLINK))
 		base_recoil -= 0.2 //Worse than kriosans but probably enough to matter
+		if(usr.stats.getPerk(PERK_STRONGARMS))
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/human/human_recoil.dm:20]: This one dosnt look correct

File: code/modules/mob/living/carbon/human/human_recoil.dm (hunk 1 above)
Comment (ThePainkiller) [reply code/modules/mob/living/carbon/human/human_recoil.dm:21]: Yeah this should've been a `-=`

**Comment by ThePainkiller** [comment]
The TL;DR is that this gives Akulas these new imprinter things that give them basically the equivalent of the human perk (which got nerfed to shit mind you) and in massive quantities, 20u of that healing chem is still literally massive for a perk. Title wasn't missleading this IS a **MASSIVE** Akula buff. Not to mention almost doubling the combat stat bonuses from the perk (no, it does not matter even if you give -999 COG/BIO/MEC, you're **pushing combat stats over a hundred** (not counting baseline stats) **through an active perk**, a thing that can be done as soon as you get out of the lift)

**Comment by Guidesu** [comment]
> The TL;DR is that this gives Akulas these new imprinter things that give them basically the equivalent of the human perk (which got nerfed to shit mind you) and in massive quantities, 20u of that healing chem is still literally massive for a perk. Title wasn't missleading this IS a **MASSIVE** Akula buff. Not to mention almost doubling the combat stat bonuses from the perk (no, it does not matter even if you give -999 COG/BIO/MEC, you're **pushing combat stats over a hundred** (not counting baseline stats) **through an active perk**, a thing that can be done as soon as you get out of the lift)

I honestly did mostly the Akula thing as a meme.
Mind you I can remove it because I purposely did that just to  see what you guys would think in which to be fair was kind of hilarious anyway back to the point.

IF you have an actual suggestion. Should I just leave the numbers where it was or tone it down due to their new bonuses? I take that criticism as a grain of salt. What solutions you bring?

File: code/modules/mob/living/carbon/human/human_recoil.dm (hunk 1 above)
Comment (Lamasmaster) [reply code/modules/mob/living/carbon/human/human_recoil.dm:21]: 0.3 is higher than 0.2 when the other one you get from marshals/cargo for steep price I dunno about that

**Comment by LiteralMushroom** [comment]
Make this just a roundstart perk PR and split any changes to existing perks into a different PR.

**Comment by Guidesu** [comment]
> Suggesting making this just a roundstart perk PR and splitting any changes to existing perks into a different PR.

Ye the akula thing was more like a meme I will remove it

**Comment by Stutternov** [comment]
I’m personally not in favor of this PR. The addition of I’m planters with no trade offs for agony reduction, painkillers, speed-boosts (which will stack with nanogate + Naramad + speed boots + hyperzine + oddity perk) is just creating more things to not use in any manner besides to powergame a character upon creation harder than we already do.

I have been vocal already about how unbalanced and ‘no trade off’ / ‘barely worth caring’ trade offs stuff has in some regards. I believe this will only make such issues worse.

**Comment by Guidesu** [comment]
> I’m personally not in favor of this PR. The addition of inplanters with no trade offs for agony reduction, painkillers, speed-boosts (which will stack with nanogate + Naramad + speed boots + hyperzine + oddity perk) is just creating more things to not use in any manner besides to powergame a character upon creation harder than we already do.
> 
> Or, in other cases, some of the perks/chems listed are racial perks or perks that already exist and this are just there to empower certain species or to stack with other perks allowing duplicate.
> 
> I have been vocal already about how unbalanced and ‘no trade off’ / ‘barely worth caring’ trade offs stuff has in some regards. I believe this will only make such issues worse.
> 
> If these implanters did something unique that other chems or perks didn’t do, and had actual trade-offs to using/taking, I would possibly reverse my opinion. But in its current state, even ignoring the Akula thing, I am not in support of this.

I going to give them the perks and a clear debuff/consequences from using it instead of being simply a straight upgrade.
Each one going to have (Aside from the actual perks that has no actual mechanical difference like Herbi and Carni and the others I going to add that have their clear debuffs and buffs) their own. Akin to uhm. Others codebase like Hestia Genemodder negative and positive points system, but instead I going to make the perks sorta of neutral so it is not good nor bad and it can be used on situations it is needed. And actually tone down a little even further the healing/speed bonus to a way lesser version (I tested it in game and it actually.... even stacking it is literally so little that)

image
image
image
image
image
image

I could make it worse.... but it is already worse..... i can make side-effects but a lot of them already have such is the purge infections that actually raises your temperature so high that you can actually take damage from it and suffer internal damage.

So you want more of this type of consequences ?

**Comment by Guidesu** [comment]
All of the chems already give a stupid high amount of NSA by using it. If you use all of them you have the risk of overloading your NSA which is already one of the side-effects.

**Comment by Trilbyspaceclone** [comment]
if it complies we can test run it for a bit

File: code/modules/racial_roundstartperks/perk_h_bravado.dm [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	color = "#ded890"
+	nerve_system_accumulations = 45
+	appear_in_default_catalog = TRUE
+	constant_metabolism = TRUE
+	scannable = TRUE
+
+/datum/reagent/medicine/hustikol/affect_blood(mob/living/carbon/M, alien, effect_multiplier)
+	M.heal_organ_damage(0.02, 0.1, 2) //Trade Healing damage to  |
+	M.add_chemical_effect(CE_BLOODCLOT, min(4,2.6)) // Clotting V
+	M.adjustOxyLoss(-0.3) // Heart start beating faster. You get more oxygen to your limbs and or gans
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/perk_h_bravado.dm:156]: Heart's not beating faste if you add CE_STABLE and then CE_PULSE -3, that's literally the heart beating **SLOWER** btw

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+	init_sight = holder.additional_darksight
+	init_flash = holder.flash_mod
+	holder.additional_darksight = -3
+	holder.flash_mod += 4
+
+/datum/perk/recklessrage/remove()
+	holder.additional_darksight = init_sight
+	holder.flash_mod = init_flash
+	..()
+/datum/perk/recklessrage/assign(mob/living/carbon/human/H)
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:23]: Why do you have two `assign()` procs separated? Wouldn't it be easier to just flatten all these modifiers into one?

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+	holder.stats.changeStat(STAT_BIO, -2)
+
+/datum/perk/recklessrage/remove()
+	holder.brute_mod_perk -= 0.10
+	holder.burn_mod_perk -= 0.10
+	holder.oxy_mod_perk -= 0.10
+	holder.toxin_mod_perk -= 0.10
+	holder.stats.changeStat(STAT_COG, 2)
+	holder.stats.changeStat(STAT_ANA, -1)
+	holder.stats.changeStat(STAT_BIO, 2)
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:40]:
```suggestion
	holder.stats.changeStat(STAT_BIO, 2)
	holder.additional_darksight += 3
	holder.flash_mod -= 4
```
Removing the perk should take off the other downsides as well

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+	return ..()
+
+/datum/reagent/stim/furiadol
+	name = "Furiadol"
+	id = "furiadol"
+	description = "A chemical naturally produced by several species that sends them into an all consuming frenzy."
+	taste_description = "bitterness"
+	reagent_state = LIQUID
+	color = "#5f95e2"
+	nerve_system_accumulations = 105 // so people can't spam it with others perks
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:65]:
```suggestion
	nerve_system_accumulations = 70 // Minimum baseline NSA for certain races/backgrounds is 80~85, this is also in par with MENACE
```
If you set it to 100 or higher any race will instantly throw up of maximum NSA threshold, granted this should still make anyone vomit if they have fucked up nerves but better than outright inducing NSA break.

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+	M.stats.addTempStat(STAT_TGH, 35, STIM_TIME, "furiadol")
+	M.stats.addTempStat(STAT_ROB, 35, STIM_TIME, "furiadol")
+	M.stats.addTempStat(STAT_COG, -300, STIM_TIME, "furiadol")
+	M.stats.addTempStat(STAT_BIO, -300, STIM_TIME, "furiadol")
+	M.stats.addTempStat(STAT_VIG, -300, STIM_TIME, "furiadol")
+	M.stats.addTempStat(STAT_MEC, -300, STIM_TIME, "furiadol")
+
+/datum/reagent/drug/furiadol/withdrawal_act(mob/living/carbon/M)
+	M.stats.addTempStat(STAT_TGH, -STAT_LEVEL_BASIC, STIM_TIME, "furiadol_w")
+	M.stats.addTempStat(STAT_ROB, -STAT_LEVEL_BASIC, STIM_TIME, "furiadol_w")
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:79]:
```suggestion
	M.stats.addTempStat(STAT_ROB, -STAT_LEVEL_BASIC, STIM_TIME, "furiadol_w")
	M.Weaken(3)
```

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+		return
+	initial_time = world.time
+	for(var/mob/living/carbon/human/H in viewers(5, holder))
+		if(H.stat == DEAD || H.internal || H.stats.getPerk(PERK_OVERWORKEDSPEED) || H.species.flags & NO_BREATHE)
+			continue
+		if(H.head?.item_flags & BLOCK_GAS_SMOKE_EFFECT || H.wear_mask?.item_flags & BLOCK_GAS_SMOKE_EFFECT || BP_IS_ROBOTIC(H.get_organ(BP_CHEST)))
+			continue
+
+		H.reagents?.add_reagent("amatoxin", 5)
+		H.emote("gasp")
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:110]:
```suggestion
		H.emote("cough")
```

Comment (ThePainkiller) [reply code/modules/racial_roundstartperks/racial_perk.dm:110]: The holder of the perk should give a span message in sight for everyone that they are releasing spores, too. Sort of like the anomaly that gives you the toxic fungus.

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+	passivePerk = FALSE
+
+/datum/perk/purgemaluses/activate()
+	var/mob/living/carbon/human/user = usr
+	if(!istype(user))
+		return ..()
+	if(world.time < cooldown_time)
+		to_chat(usr, SPAN_NOTICE("Your body aches with the pain of its recent purge, you'll need more rest before using this effect again."))
+		return FALSE
+	cooldown_time = world.time + 15 MINUTES
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:160]:
```suggestion
	cooldown_time = world.time + 30 MINUTES
```

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+/datum/perk/purgemaluses/assign(mob/living/carbon/human/H)
+	..()
+	holder.toxin_mod_perk += 0.1 //Might be to high...
+
+/datum/perk/purgemaluses/remove()
+	holder.toxin_mod_perk -= 0.1
+	..()
+
+
+/datum/reagent/medicine/spaceacillin/sataricillin
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:175]: This whole medicine is acting 1:1 with spaceacillin and does not purge any toxins, nor stimulants, nor reduces your toxloss in any way, this is just a 5u shot of Spaceacillin. I suggest adding its own affect_blood() proc with the combined effects of:
**Purger**
 `M.add_chemical_effect(CE_PURGER, 1)`
**Haloperidol since it removes Toxins and Stims:**
```	if(M.bloodstr)
		for(var/current in M.bloodstr.reagent_list)
			var/datum/reagent/toxin/R = current
			if(!istype(R, src))
				R.remove_self(effect_multiplier * pick(list(2,4,6)))
			var/datum/reagent/stim/S = current
			if(!istype(S, src))
				S.remove_self(effect_multiplier * pick(list(2,4,6)))
```
**And some minor Dylovene effect:**
`M.adjustToxLoss(-((0.1 + (M.getToxLoss() * 0.05)) * effect_multiplier))`

As well as upping the cooldown to 30 minutes for the perk as suggested earlier since that'd make it have the desired effects and would be too good to spam every fifteen.

File: code/modules/racial_roundstartperks/racial_perk.dm [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+	if(world.time < cooldown_time)
+		to_chat(usr, SPAN_NOTICE("Your chemical sacks have not refilled yet, you'll need more rest before using this effect again."))
+		return FALSE
+	cooldown_time = world.time + 60 MINUTES
+	user.visible_message("[user] shivers slightly before taking a deep breath.... they look hot... are they in a heavy fever?", "You shiver slightly and take a deep breath before willing your bodies chemical sacks to open and begin purging infections.")
+	log_and_message_admins("used their [src] perk.")
+	user.reagents.add_reagent("citokinesimol", 5)
+	return ..()
+
+/datum/reagent/medicine/citokinesimol
```
Comment (ThePainkiller) [inline code/modules/racial_roundstartperks/racial_perk.dm:211]: This...chem is doing the exact same stuff I suggested on the earlier chem, making these two perks pretty much the same with almost no difference, **why**?

**Review by ThePainkiller — CHANGES_REQUESTED** [review]
While these are more balanced now with the latest commits, there's still room for improvement

**Comment by Trilbyspaceclone** [comment]
I want to run some local tests on this and clean up some code stuff before merging.

**Comment by REDLAERT88** [comment]
Right, after discussing this with a few other staff, and getting opinions I'm denying this PR on lore reasons. 

The damage done to the uniqueness of our existing species is not something that can be balanced around. The idea of a kriosan popping a cinderite based perk (though heavily altered and changed) is not conductive in any way to the unique feeling of each species.

My comments for how to correct and make this system workable, will be to. 

A. All "roundstart" perk injectors MUST be NEW content and have NO overlap with existing species abilities as to not harm the uniqueness of the existing races. (This is the key main reason for the denial at this time as the way it is coded)

B. The price of the perk injectors MUST be increased on the grounds that these are ROUND LASTING PERKS. 2 loadout points is nowhere near enough for a permeant round long ability. Most species only get 2-3 perks, giving FIVE perks in loadout seems excessive to me in terms of reasonable modification of ones self. 

C. The fact these are injectors and loadout items in the first place is also questionable, as one could spawn with a bunch with the pure intent to give them or trade them to others, which doesn't seem to be the intended use for this system. This system would benefit much more by being a background tab option, under the idea of gene modding, rather than injecting yourself every round.

---

The INTENT of this system is understandable and useable, it is a good idea, however this execution isn't acceptable due to the damage it may, and/or will cause.

**Comment by Guidesu** [comment]
> Right, after discussing this with a few other staff, and getting opinions I'm denying this PR on lore reasons.
> 
> The damage done to the uniqueness of our existing species is not something that can be balanced around. The idea of a kriosan popping a cinderite based perk (though heavily altered and changed) is not conductive in any way to the unique feeling of each species.
> 
> My comments for how to correct and make this system workable, will be to.
> 
> A. All "roundstart" perk injectors MUST be NEW content and have NO overlap with existing species abilities as to not harm the uniqueness of the existing races. (This is the key main reason for the denial at this time as the way it is coded)
> 
> B. The price of the perk injectors MUST be increased on the grounds that these are ROUND LASTING PERKS. 2 loadout points is nowhere near enough for a permeant round long ability. Most species only get 2-3 perks, giving FIVE perks in loadout seems excessive to me in terms of reasonable modification of ones self.
> 
> C. The fact these are injectors and loadout items in the first place is also questionable, as one could spawn with a bunch with the pure intent to give them or trade them to others, which doesn't seem to be the intended use for this system. This system would benefit much more by being a background tab option, under the idea of gene modding, rather than injecting yourself every round.
> 
> The INTENT of this system is understandable and useable, it is a good idea, however this execution isn't acceptable due to the damage it may, and/or will cause.

It's easier said than done.
😕

**Comment by Guidesu** [comment]
going work on it tomorrow

**Comment by cdb-is-not-good** [comment]
Don't make them loadout items at all, that's not really suitable. The loadout system has been very much adjusted such that you rarely if ever will be spending many points, because YOU specifically Guidesu lowered the amount of points every clothing item costed. Make it a background item is probably the best way to go as Cora suggested.

**Comment by Guidesu** [comment]
#4178
I going to merge this with the lamas PR that he closed into my own thing

**Comment by Guidesu** [comment]
> Don't make them loadout items at all, that's not really suitable. The loadout system has been very much adjusted such that you rarely if ever will be spending many points, because YOU specifically Guidesu lowered the amount of points every clothing item costed. Make it a background item is probably the best way to go as Cora suggested.

Going try to work in a better system
